std/format/package.d

```diff
@@ -1931,13 +1941,43 @@ char[] sformat(Char, Args...)(return scope char[] buf, scope const(Char)[] fmt,
     return buf[0 .. sink.i];
 }
 
-/// The format string can be checked at compile-time (see $(LREF format) for details):
+/// ditto
+char[] sformat(alias fmt, Args...)(char[] buf, Args args)
+if (isSomeString!(typeof(fmt)))
+{
+    alias e = checkFormatException!(fmt, Args);
+    static assert(!e, e.msg);
+    return .sformat(buf, fmt, args);
+}
+
+///
+@safe pure unittest
+{
+    char[20] buf;
+    assert(sformat(buf[], "Here are %d %s.", 3, "apples") == "Here are 3 apples.");
+
+    assert(buf[].sformat("Increase: %7.2f %%", 17.4285) == "Increase:   17.43 %");
+}
+
+/// The format string can be checked at compile-time:
+@safe pure unittest
+{
+    char[20] buf;
+
+    assert(sformat!"Here are %d %s."(buf[], 3, "apples") == "Here are 3 apples.");
+
+    // This line doesn't compile, because 3.14 cannot be formatted with %d:
+    // writeln(sformat!"Here are %d %s."(buf[], 3.14, "apples"));
```

Comment:
It looks kind of bad to have commented code in public examples. Will the unittest become `@system` if we replace this comment with: `static assert(!__traits(compiles, sformat!"Here are %d %s."(buf[], 3.14, "apples")));` ?

Reply:
Using `__traits` is possible, but I dislike it. For a beginner, I think, it's much more comprehensible to have commented code, than a `__traits` statement. (Note, that I added this `__traits` statement in the next (nonpublic) unittest, to actually verify this statement.)

Whatever we decide upon here, we should do the same for the similar example of `format` (a PR that was already merged yesterday).

Reply:
Fair enough.